### PR TITLE
Added Windows Tests

### PR DESCRIPTION
**Pull request by TatianaKapos**
# Summary
Added Windows tests to ProgressView

- Since circleCI has issues with windows, used Github actions to create the windows tests.
- Test simply makes sure each progressbar can be found, will time out and fail if the progressbar does not render and will tell you the specific progress bar it failed at.

image

## Checklist

- [x] I have tested this on a device and a simulator
- [x] I added the documentation in `README.md`
- [ ] I mentioned this change in `CHANGELOG.md`
- [ ] I updated the typed files (TS and Flow)
- [x] I added a sample use of the API in the example project (`example/App.js`)

### CODE REVIEW

File: .github/workflows/windows-ci.yml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+    - name: Build x64 release
+      run: msbuild .\example\windows\ProgressViewExample.sln /p:Configuration=Release /p:Platform=x64 -m
+
+    - name: Deploy
+      shell: powershell
+      run: |
+        cd example
+        Copy-Item -Path windows\x64\Release -Recurse -Destination windows\
+        npx react-native run-windows --arch x64 --release --no-build --no-packager
```
Comment (asklar) [inline .github/workflows/windows-ci.yml:57]: why is this task needed? 🤔

File: example/.gitignore [hunk 2]
```diff
@@ -54,3 +54,6 @@ buck-out/
 
 # Bundle artifact
 *.jsbundle
+
+msbuild.binlog
```
Comment (asklar) [inline example/.gitignore:58]: you can just have *.binlog at the top instead since we would never want to check this file type in anyway

File: windows/progress-view/ProgressViewView.cpp [hunk 3]
```diff
@@ -71,6 +71,9 @@ namespace winrt::progress_view::implementation {
                     this->Background(imgB);
                 }
             }
+            else if (propertyName == "nativeName") {
+                this->Name(to_hstring(propertyValue.AsString()));
+            }
```
Comment (asklar) [inline windows/progress-view/ProgressViewView.cpp:76]: is this a custom prop? if so see if we can avoid it by using `testID` instead

Comment (TatianaKapos) [reply windows/progress-view/ProgressViewView.cpp:76]: I know I played around with getting testID to work but it wouldn't show up from some reason on the UWP side, so this was my work around. I'll play around with testID some more and see what specific error I was getting!

File: .github/workflows/windows-ci.yml (hunk 1 above)
Comment (kaiguo) [reply .github/workflows/windows-ci.yml:57]: I think it deploys the package and launches the app.

Comment (TatianaKapos) [reply .github/workflows/windows-ci.yml:57]: that sounds right! I just tried removing it and it gives me a "Windows cannot perform the RegisterByPackageFamilyName operation because it could not find ProgressViewExample_kvqejeg71e612 in the repository." error.

File: windows/progress-view/ProgressViewView.cpp (hunk 3 above)
Comment (TatianaKapos) [reply windows/progress-view/ProgressViewView.cpp]: fixed! Looks like I was setting up View props incorrectly

File: .eslintignore [hunk 4]
```diff
@@ -1 +1,4 @@
-node_modules
\ No newline at end of file
+node_modules
+jest-setups/jest.setup.js
+jest-setups/jest.setup.windows.js
+__tests__/ProgressViewWindows.test.js
```
Comment (asklar) [inline .eslintignore:4]: is it necessary to ignore this file?

File: .github/workflows/windows-ci.yml [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+    - name: Build x64 release
+      run: msbuild .\example\windows\ProgressViewExample.sln /p:Configuration=Release /p:Platform=x64 -m
+      
+    - name: Deploy
+      shell: powershell
+      run: |
+        cd example
+        Copy-Item -Path windows\x64\Release -Recurse -Destination windows\
+        npx react-native run-windows --arch x64 --release --no-build --no-packager
```
Comment (asklar) [inline .github/workflows/windows-ci.yml:57]: sorry let me rephrase: can we just replace all of this with just running yarn windows or npx react-native run-windows?

File: example/windows/ProgressViewExample/ProgressViewExample.vcxproj [hunk 6]
```diff
@@ -153,6 +155,12 @@
     </ProjectReference>
   </ItemGroup>
   <Import Project="$(VCTargetsPath)\Microsoft.Cpp.targets" />
+  <PropertyGroup>
+    <BundleEntryFile>index.js</BundleEntryFile>
+    <BundlerExtraArgs>--use-react-native-windows</BundlerExtraArgs>
+  </PropertyGroup>
+  <Import Project="..\..\..\node_modules\react-native-windows\PropertySheets\Bundle.Cpp.targets" />
+  <Import Project="..\..\..\node_modules\react-native-windows\PropertySheets\Bundle.props" />
```
Comment (asklar) [inline example/windows/ProgressViewExample/ProgressViewExample.vcxproj:163]: import props before targets

File: .github/workflows/windows-ci.yml (hunk 5 above)
Comment (kaiguo) [reply .github/workflows/windows-ci.yml:57]: I remember seeing some bundling errors when building release using just run-windows command, maybe that's fixed now. @TatianaKapos you can try replacing all this with `npx react-native run-windows --arch x64 --release --no-packager` and see if it works.

Comment (TatianaKapos) [reply .github/workflows/windows-ci.yml]: hmmm yeah I'm seeing an error where it can't find the package in the release folder :/

File: .eslintignore (hunk 4 above)
Comment (TatianaKapos) [reply .eslintignore:4]: I removed the test file from this file but both jest-setup files gives me a `'platform' is not defined  no-undef` error even though it's defined in the setup.windows

**Comment by TatianaKapos** [comment]
@Naturalclar This PR had been reviewed, let me know if you need me to make any changes! :)

**Comment by chrisglein** [comment]
I was looking to add the Windows state to [reactnative.directory](https://reactnative.directory/), but it appears that we need a new published update that this contains this merged PR. In particular to address the part to fix the package files that was originally a [separate PR](https://github.com/react-native-community/progress-view/pull/39) but then was included in this PR.

The change is in, there just needs to be a new publish to NPM. @Naturalclar are releases being done on demand? What's the criteria for pushing out a new one?

@TatianaKapos Did we get validation that if the `windows` folder is published then the Windows version is all good?

**Comment by Naturalclar** [comment]
@chrisglein I can release the package on demand. I thought the PR only added test and no change to the actual functionality of the code, so I didn't make a release, sorry about that!
I'll make a new release soon

**Comment by TatianaKapos** [comment]
My bad, should have specified the fix in the PR! But I just checked and it fixes the windows issue.

**Comment by asklar** [comment]
@TatianaKapos we miss youuuuuuu
